cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/SystemArguments.java

```diff
@@ -50,38 +51,36 @@
   private static final String RETRY_POLICY_DELAY_MAX_MS = "system." + Constants.Retry.DELAY_MAX_MS;
   public static final String TRANSACTION_TIMEOUT = "system.data.tx.timeout";
 
-  public static Map<String, String> getLogLevels(Map<String, String> args) {
-    Map<String, String> logLevels = new HashMap<>();
+  /**
+   * Extracts log level settings from the given arguments. It extracts arguments prefixed with key
+   * {@link #LOG_LEVEL} + {@code .}, with the remaining part of the key as the logger name, with the argument value
+   * as the log level. Also, the key {@link #LOG_LEVEL} will be used to setup the log level of the root logger.
+   */
+  public static Map<String, Level> getLogLevels(Map<String, String> args) {
+    String logLevelPrefix = LOG_LEVEL + ".";
+
+    Map<String, Level> logLevels = new HashMap<>();
     for (Map.Entry<String, String> entry : args.entrySet()) {
-      String loggerName = entry.getKey();
-      if (loggerName.length() > LOG_LEVEL.length() && loggerName.startsWith(LOG_LEVEL)) {
-        logLevels.put(loggerName.substring(LOG_LEVEL.length() + 1), entry.getValue());
+      String key = entry.getKey();
+      if (key.startsWith(logLevelPrefix)) {
+        logLevels.put(key.substring(logLevelPrefix.length()), Level.toLevel(entry.getValue()));
       }
     }
     String logLevel = args.get(LOG_LEVEL);
```

Comment:
hmmm, you changed this to include the ".", this will not work any longer...

Reply:
ROOT logger is set with `system.log.level` (which is what `LOG_LEVEL` defined to), while specific logger is set with `system.log.level.X`.